c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -1581,6 +1592,11 @@ class DeviceCachingAllocator {
         continue;
       }
 
+      // unmapped blocks in the checkpoint state are already in the free pool
```

Comment:
I don't think this code is reachable: since cuda graph memory cannot be freed, we cannot end up with an unmapped block that is in the middle of a segment, but there is already a skip above. 

If it turns out this code _is_ reachable, then it isn't quite correct: what if this memory _has been mapped_ since the time the checkpoint has taken? Then the next allocation is going to get restored here rather than in the place we expect it to.

Reply:
The restoring checkpoint logic does two passes over all the blocks, first to allocate all allocated blocks in the segment then to free unallocated blocks. the last block in an expandable segment is unmapped so we need checks to skip over allocating/freeing the unmapped blocks which is what this was meant to do.

I'll add over the same checks from above here just to make it match.